specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -42,16 +43,52 @@ export function Notifications({
 
   React.useEffect(() => {
     let pullInterval = INITIAL_INTERVAL;
+    let lastFetchedTimestamp: Date | undefined;
+
     const handler = (): void => {
-      if (timeout !== undefined) globalThis.clearTimeout(timeout);
+      if (timeout !== undefined) {
+        globalThis.clearTimeout(timeout);
+      }
       pullInterval = INITIAL_INTERVAL;
-      if (document.visibilityState === 'visible') doFetch();
+      if (document.visibilityState === 'visible') {
+        doFetch();
+      }
     };
+
     document.addEventListener('visibilitychange', handler);
 
     let timeout: NodeJS.Timeout | undefined = undefined;
 
-    function doFetch(): void {
+    function doFetch(since?: Date): void {
+      const startFetchTimestamp = new Date();
+
+      const now = new Date();
```

Comment:
does `new Date().toISOString().replace('T', ' ')` or `new Date().toJSON().replace('T', ' ')` not work?
it works in other places where front-end communicates with back-end so should work here too - and would be cleaner than this

or at very least extract this logic into some sort of function! don't you see how much code duplication there is here? 🫨

Reply:
take some time to do research on how the front-end currently sends dates to the back-end in other places (i.e when you set a `cataloggedDate` field on the collection object - will be helpful for you both for this issue and for future

Reply:
Unfortunately no, new Date().toISOString().replace('T', ' ') or new Date().toJSON().replace('T', ' ') do not work

Reply:
I tried: 
- Date().toISOString().replace('T', ' ') 
- new Date().toJSON().replace('T', ' ') 
- encodeURIComponent(new Date().toISOString())
- new Date().toISOString()
- new Date().toJSON()
- `${new Date().getFullYear()}-${new Date().getMonth()}-${new Date().getDay()} ${new Date().getHours()}:${new Date().getMinutes()}:${new Date().getSeconds()}}
          `
- new Date().toISOString().replace('T', ' ')
- const now = new Date();
const formattedDate = now.toLocaleString();

Having errors with all of those I tested with the long version using get(...